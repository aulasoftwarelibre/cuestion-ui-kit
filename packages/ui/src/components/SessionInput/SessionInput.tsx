import TextField from "@material-ui/core/TextField";

import * as React from "react";
import { InjectedIntl, injectIntl } from "react-intl";

import { messages } from "@cuestion/common";

interface Props {
  handleOnSubmit: any;
  intl: InjectedIntl;
  length: number;
}

const _SessionInput: React.FunctionComponent<Props> = ({
  handleOnSubmit,
  intl,
  length,
}) => {
  const [code, setCode] = React.useState<string>("");

  const handleChange = () => (event: React.ChangeEvent<HTMLInputElement>) => {
    setCode(
      event.target.value
        .toUpperCase()
        .substring(0, length)
        .replace(/[^A-Z0-9]+/g, ""),
    );
  };

  const onSubmit = (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    if (code.length === length) {
      handleOnSubmit(code);
      setCode("");
    }
  };

  const messageEnter = intl.formatMessage(messages.pressEnter, {});
  const messageEnterLengthCharactersCode = intl.formatMessage(
    messages.enterLengthCharactersCode,
    {
      length,
    },
  );
  const messageSessionCode = intl.formatMessage(messages.sessionCode, {});

  return (
    <form onSubmit={onSubmit}>
      <TextField
        id="session-code-input"
        name="code"
        type="text"
        label={messageSessionCode}
        helperText={
          code.length === length
            ? messageEnter
            : messageEnterLengthCharactersCode
        }
        onChange={handleChange()}
        value={code}
        variant="outlined"
      />
    </form>
  );
};

export const SessionInput = injectIntl(_SessionInput);

export default SessionInput;
