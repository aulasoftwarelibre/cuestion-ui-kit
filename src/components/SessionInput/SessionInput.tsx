import TextField from "@material-ui/core/TextField";
import * as React from "react";
import { InjectedIntl, injectIntl } from "react-intl";

import messages from "../../languages/messages";

interface Props {
  handleSubmit: any;
  intl: InjectedIntl;
  length?: number;
}

const SessionInput: React.FunctionComponent<Props> = ({
  handleSubmit,
  intl,
  length = 4
}) => {
  const [code, setCode] = React.useState<string>("");

  const handleChange = () => (event: React.ChangeEvent<HTMLInputElement>) => {
    setCode(
      event.target.value
        .toUpperCase()
        .substring(0, length)
        .replace(/[^A-Z0-9]+/g, "")
    );
  };

  const onSubmit = (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    const sessionCode: string = e.target[1].value;
    if (sessionCode.length === length) {
      handleSubmit(sessionCode);
      setCode("");
    }
  };

  return (
    <form onSubmit={onSubmit}>
      <TextField
        id="session-code-input"
        name="code"
        type="text"
        label={intl.formatMessage(messages.sessionCode, {})}
        helperText={
          code.length === length
            ? intl.formatMessage(messages.pressEnter, {})
            : intl.formatMessage(messages.enterLengthCharactersCode, { length })
        }
        onChange={handleChange()}
        value={code}
        variant="outlined"
      />
    </form>
  );
};

export default injectIntl(SessionInput);
