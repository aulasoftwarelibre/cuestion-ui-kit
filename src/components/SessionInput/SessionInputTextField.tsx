import TextField from "@material-ui/core/TextField";
import * as React from "react";

interface Props {
    handleOnSubmit: any;
    length: number;
    messageEnter: string;
    messageEnterLengthCharactersCode: string;
    messageSessionCode: string;
}

const SessionInputTextField: React.FunctionComponent<Props> = ({
    handleOnSubmit,
    length,
    messageEnter,
    messageEnterLengthCharactersCode,
    messageSessionCode
}) => {
    const [code, setCode] = React.useState<string>("");

    const handleChange = () => (event: React.ChangeEvent<HTMLInputElement>) => {
        setCode(
            event.target.value
                .toUpperCase()
                .substring(0, length)
                .replace(/[^A-Z0-9]+/g, "")
        );
    };

    const onSubmit = (e: React.FormEvent<HTMLFormElement>) => {
        e.preventDefault();
        const sessionCode: string = e.target[1].value;
        if (sessionCode.length === length) {
            handleOnSubmit(sessionCode);
            setCode("");
        }
    };

    return (
        <form onSubmit={onSubmit}>
            <TextField
                id="session-code-input"
                name="code"
                type="text"
                label={messageSessionCode}
                helperText={
                    code.length === length
                        ? messageEnter
                        : messageEnterLengthCharactersCode
                }
                onChange={handleChange()}
                value={code}
                variant="outlined"
            />
        </form>
    );
};

export default SessionInputTextField;
