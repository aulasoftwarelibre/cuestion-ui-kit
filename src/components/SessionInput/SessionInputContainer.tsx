import * as React from "react";
import { InjectedIntl, injectIntl } from "react-intl";

import messages from "../../languages/messages";
import SessionInput from "./SessionInput";

interface Props {
  handleOnSubmit: any;
  intl: InjectedIntl;
  length: number;
}

class SessionInputContainer extends React.Component<Props> {
  constructor(props: Props) {
    super(props);
  }

  public render() {
    const { handleOnSubmit, intl, length } = this.props;

    return (
      <SessionInput
        handleOnSubmit={handleOnSubmit}
        length={length}
        messageEnter={intl.formatMessage(messages.pressEnter, {})}
        messageEnterLengthCharactersCode={intl.formatMessage(messages.enterLengthCharactersCode, { length })}
        messageSessionCode={intl.formatMessage(messages.sessionCode, {})}
      />
    );
  }
}

export default injectIntl(SessionInputContainer);
