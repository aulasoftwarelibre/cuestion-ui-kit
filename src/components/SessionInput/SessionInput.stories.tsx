import { action } from "@storybook/addon-actions";
import { number, withKnobs } from "@storybook/addon-knobs";
import { storiesOf } from "@storybook/react";
import * as React from "react";

import { wInfo } from "../../utils/wInfo";
import SessionInputContainer from "./SessionInputContainer";

const stories = storiesOf("Components/SessionInput", module);
stories.addDecorator(withKnobs);

stories.add(
  "default",
  wInfo()(() => (
    <SessionInputContainer
      handleOnSubmit={action("session-code")}
      length={number("Length", 4, { range: true, min: 4, max: 8, step: 1 })}
    />
  ))
);
