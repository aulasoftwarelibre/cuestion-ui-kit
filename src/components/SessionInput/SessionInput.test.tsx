import * as React from "react";

import {
  rendererWithIntl,
  shallowWithIntl
} from "../../helpers/intl-enzyme-test-helper";
import SessionInput from "./SessionInput";

describe("SessionInput", () => {
  const mockHandleSubmit = jest.fn();

  it("should render correctly", () => {
    const component = rendererWithIntl(
      <SessionInput handleSubmit={mockHandleSubmit} />
    ).toJSON();

    expect(component).toMatchSnapshot();
  });

  it("should tell to press ENTER", () => {
    const wrapper = shallowWithIntl(
      <SessionInput handleSubmit={mockHandleSubmit} />
    );
    wrapper.find("#session-code-input").simulate("change", {
      target: { name: "session-code-input", value: "1234" }
    });

    expect(
      wrapper.find("#session-code-input").prop("helperText")
    ).toStrictEqual("and press ENTER.");
  });

  it("should tell to wait to fill code", () => {
    const wrapper = shallowWithIntl(
      <SessionInput length={6} handleSubmit={mockHandleSubmit} />
    );
    wrapper.find("#session-code-input").simulate("change", {
      target: { name: "session-code-input", value: "1234" }
    });

    expect(
      wrapper.find("#session-code-input").prop("helperText")
    ).toStrictEqual("Enter 6 characters code.");
  });
});
