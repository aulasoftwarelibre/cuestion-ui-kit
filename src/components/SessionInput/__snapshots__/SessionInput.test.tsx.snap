// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SessionInput should render correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="MuiFormControl-root MuiTextField-root"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
      data-shrink={false}
      htmlFor="session-code-input"
    >
      <span>
        Session Code
      </span>
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
      onClick={[Function]}
    >
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-57 MuiOutlinedInput-notchedOutline"
        style={
          Object {
            "paddingLeft": 8,
          }
        }
      >
        <legend
          className="PrivateNotchedOutline-legend-58"
          style={
            Object {
              "width": 0.01,
            }
          }
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </legend>
      </fieldset>
      <input
        aria-describedby="session-code-input-helper-text"
        aria-invalid={false}
        className="MuiInputBase-input MuiOutlinedInput-input"
        disabled={false}
        id="session-code-input"
        name="code"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
        value=""
      />
    </div>
    <p
      className="MuiFormHelperText-root MuiFormHelperText-contained"
      id="session-code-input-helper-text"
    >
      Enter 4 characters code.
    </p>
  </div>
</form>
`;
